tendermint/src/rpc/endpoint/abci_query.rs

```diff
@@ -69,24 +71,40 @@ pub struct AbciQuery {
     /// Index
     #[serde(
         serialize_with = "serializers::serialize_i64",
-        deserialize_with = "serializers::parse_i64"
+        deserialize_with = "serializers::parse_i64",
+        default
     )]
     pub index: i64,
 
     /// Key
     // TODO(tarcieri): parse to Vec<u8>?
-    pub key: String,
+    pub key: Option<String>,
 
     /// Value
     // TODO(tarcieri): parse to Vec<u8>?
-    pub value: String,
+    pub value: Option<String>,
 
     /// Proof (if requested)
     pub proof: Option<Proof>,
 
     /// Block height
-    pub height: block::Height,
+    #[serde(default, deserialize_with = "serializers::parse_height_option")]
+    pub height: Option<block::Height>,
 
     /// Codespace
     pub codespace: Option<String>,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json;
+
+    #[test]
+    fn parse_query_responses() {
+        let error_rsp = r#"{"response": {"code": 1, "log": "account lookup failed: account not found", "info": "", "index": "0", "key": null, "value": null, "proof": null, "height": "0", "codespace": ""}}"#;
+        let success_rsp = r#"{"response": {"value": "some value"}}"#;
```

Comment:
Doesn't a successful response also include `code` etc? Let me double check.

Reply:
It seems it depends solely on the abci app's implementation, and since it's protobuf, every field is optional.

Reply:
It might be good to update our fixtures in the rpc to contain both JSON files, the happy path and the error case:
https://docs.tendermint.com/master/rpc/#/ABCI/abci_query